### ISSUE TITLE
Validation Batch size

### ISSUE THREAD

**Issue by jaideep11061982** [open-mmlab/mmsegmentation#1645]
I find that the validation set is evaluated image/image which takes lot of time. I even specifiy the samples per gpu as 32.

**data = dict(
    samples_per_gpu=32,
    workers_per_gpu=4,**

**runner = dict(type='IterBasedRunner', max_iters=int(total_iters * 1000))**

**Comment by jaideep11061982** [comment]
evaluates  image by image by looking at the progress bar of evaluation it appears to be.

**Comment by MengzhangLI** [comment]
Because `samples_per_gpu` only works for model training, and the default setting of `samples_per_gpu` is 1 in mmseg when model testing and validation (DO NOT support batch inference yet).

So if you want to speed up right now, you could only use more GPU training/testing.

**Comment by jaideep11061982** [comment]
@MengzhangLI thanku
what is difference between Iter based runner and Epoch based. 
If i want to run for say 9 epochs , then no  of iterations will be   (ds length/sampler_per_gpu) * no of epochs ?

**Comment by MengzhangLI** [comment]
You can calculate it by relationship `iterations = epochs * total batchsize`.
